pkg/expect/expect.go

```diff
@@ -44,7 +46,15 @@ var printDebugLines = os.Getenv("EXPECT_DEBUG") != ""
 
 // NewExpect creates a new process for expect testing.
 func NewExpect(name string, arg ...string) (ep *ExpectProcess, err error) {
-	ep = &ExpectProcess{cmd: exec.Command(name, arg...)}
+	// if env[] is nill, use current system env
+	return NewExpectWithEnv(name, nil, arg...)
+}
+
+// NewExpectWithEnv creates a new process with user defined env variables for expect testing.
+func NewExpectWithEnv(name string, env []string, arg ...string) (ep *ExpectProcess, err error) {
```

Comment:
this is not the conventional order-- it should be `NewExpectWithEnv(name string, args[] string, env[] string)` to match `execvpe`